includes/reader-revenue/stripe/class-stripe-sync.php

```diff
@@ -454,36 +520,139 @@ protected static function process_customer_for_stripe_connect_to_stripe( $custom
+							\WP_CLI::log(
+								sprintf(
+									'    * Would update WC Subscription #%s, which was already synced from Stripe but not renewed by WC.',
+									$subscription_id
+								)
+							);
+						} else {
+							WooCommerce_Connection::add_wc_stripe_gateway_metadata( $subscription, $stripe_metadata_base );
+							// Delete the 'link' to a Stripe subscription, so it's not sync'd anymore.
+							$subscription->delete_meta_data( WooCommerce_Connection::SUBSCRIPTION_STRIPE_ID_META_KEY );
+							$subscription->save();
+							\WP_CLI::success( sprintf( 'Updated WC Subscription #%s, which was already synced from Stripe but not renewed by WC (now it will).', $subscription_id ) );
+						}
+					} else {
+						// Create a new WooCommerce subscription, starting at the current period start.
+						$currency                = $existing_subscription->currency;
+						$amount_normalised       = Stripe_Connection::normalise_amount( $existing_subscription->quantity, $currency );
+						$first_subscription_item = $existing_subscription->items->data[0];
+						$frequency               = $first_subscription_item->price->recurring->interval;
+
+						$wc_order_payload = array_merge(
+							[
+								'status'                 => 'completed',
+								'subscription_status'    => 'created',
+								'wc_subscription_status' => 'active',
+								'email'                  => $customer->email,
+								'name'                   => $customer->name,
+								'date'                   => $existing_subscription->current_period_start,
+								'amount'                 => $amount_normalised,
+								'frequency'              => $frequency,
+								'currency'               => strtoupper( $currency ),
+								'client_id'              => $customer->metadata->clientId,
+								'user_id'                => $customer->metadata->userId,
+							],
+							$stripe_metadata_base
+						);
+
+						if ( $dry_run ) {
+							\WP_CLI::log(
+								sprintf(
+									'    * Would create a WC Subscription with frequency of %s and amount of %s.',
+									$frequency,
+									$amount_normalised
+								)
+							);
+							$subscription_id = true;
+						} else {
+							// Create the WC Subscription.
+							$wc_transaction_creation_data = WooCommerce_Connection::create_transaction( $wc_order_payload );
+							$subscription_id              = $wc_transaction_creation_data['subscription_id'];
+							$subscription                 = \wcs_get_subscription( $subscription_id );
+							\WP_CLI::success( sprintf( 'Created subscription: %s with next renewal at %s', $subscription_id, gmdate( 'Y-m-d', $existing_subscription->current_period_end ) ) );
+						}
+					}
+
+					if ( $subscription ) {
+						// Add the cancelled Stripe subscription ID to the meta data, so it can be found later.
+						$subscription->add_meta_data( 'cancelled-' . WooCommerce_Connection::SUBSCRIPTION_STRIPE_ID_META_KEY, $existing_subscription->id );
+						$subscription->add_order_note(
```

Comment:
This note is added to orders on dry runs, too. We should probably only add the note if it's not a dry run.

Reply:
Thanks for catching that! Fixed in cdfed16e218b91491ce98f093152ec665ceacc03